pkg/slayers/scion.go

```diff
@@ -553,19 +553,39 @@ func scionNextLayerTypeL4(t L4ProtocolType) gopacket.LayerType {
 // from the underlaying layer data. Changing the net.Addr object might lead to inconsistent layer
 // information and thus should be treated read-only. Instead, SetDstAddr should be used to update
 // the destination address.
-// @ trusted
-// @ requires false
-func (s *SCION) DstAddr() (net.Addr, error) {
+// @ requires  acc(&s.DstAddrType) && acc(&s.RawDstAddr)
+// @ requires  s.DstAddrType.Has3Bits()
```

Comment:
will we be able to establish this property in the call site?

Reply:
I certainly hope so, because without this we can't prove that there won't be an out of bounds reslicing in parseAddr